### ISSUE TITLE
Is it possible to get the starting/ending index of an defined array?

### ISSUE THREAD

**Issue by EduardoVaz06** [integrated-application-development/sonar-delphi#257]
### Prerequisites

- [X] This question has not already been answered.
- [X] This question is about SonarDelphi, not SonarQube or Delphi.

### Inquiry

Hi!

I was trying to create a rule that checks if an array is created defining an starting index other then 0 (unwanted in my company business rule), like arMyArray: array[2..5] of Integer.

The thing is, I was able to get the array name and type (ArrayTypeImpl@116 "array of System.Integer <FIXED>") but couldn't in any way get the starting/ending index of it. Is there a way for doing this in Sonar Delphi?

One thought about doing this rule was samehow getting the array declaration as a string, and checking if it contains array[X>0...] or something like that, but I don't think it is the best option, only as a plan B if there is no how to actually get the index.

Thank!

**Comment by zaneduffield** [comment]
This looks like an area for improvement in the API for sure.

The best you can do now is look at how the grammar constructs the nodes here
https://github.com/integrated-application-development/sonar-delphi/blob/a3814fedceefe55ffd604e66b0262107c539b94d/delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g#L409-L419

So you should be able to do something like
```java
@Override
public DelphiCheckContext visit(ArrayTypeNode node, DelphiCheckContext data) {
  var indices = node.getFirstChildOfType(ArrayIndicesNode.class);
  var range = indices.getFirstChildOfType(SubRangeTypeNode.class);
  var low = range.getLowExpression();
  var high = range.getHighExpression();
  
  return super.visit(node, data);
}
```

I find it really helps to set a breakpoint with a debugger and inspect the AST nodes.

**Comment by EduardoVaz06** [comment]
Got it, thank you very much!

Sometime soon I'll open an issue with an API improvement solicitation since you agree this is an area that could be improved, but for now this helps in what I wanted.
